### PR TITLE
Sunspot implementation

### PR DESCRIPTION
**Pull request by architvasan**

### CODE REVIEW

File: mdensemble/utils.py [hunk 1]
```diff
@@ -49,4 +49,5 @@ def dump_yaml(self, filename: PathLike) -> None:
     def from_yaml(cls: Type[_T], filename: PathLike) -> _T:
         with open(filename) as fp:
             raw_data = yaml.safe_load(fp)
+        print(raw_data)
```
Comment (braceal) [inline mdensemble/utils.py]: Can we remove this print?

File: mdensemble/parsl.py [hunk 2]
```diff
@@ -149,5 +149,77 @@ def config_factory(self, run_dir: PathLike) -> Config:
...
+                    cores_per_worker=16,
+                    heartbeat_period=30,
+                    heartbeat_threshold=300,
+                    worker_debug=False,
+                    provider=PBSProProvider(
+                        launcher=MpiExecLauncher(
+                            bind_cmd="--cpu-bind",
+                            overrides="--depth=208 --ppn 1"
+                        ),  # Ensures 1 manger per node and allows it to divide work among all 208 threads
+                        #worker_init="""
```
Comment (braceal) [inline mdensemble/parsl.py]: Are these flags no longer needed?

File: examples/example.yaml [hunk 3]
```diff
@@ -18,23 +18,23 @@ simulation_config:
...
-#   queue: debug
-#   walltime: 00:30:00
+compute_settings:
+   name: sunspot # Run on the Polaris supercomputer @ ALCF
+   num_nodes: 1 # 4 GPUs per node, if you change this, make sure to change num_parallel_tasks
+   worker_init: "module load frameworks/2024.1; conda activate /lus/gila/projects/candle_aesp_CNDA/avasan/envs/mdensemble"
+   scheduler_options: "#PBS -l filesystems=home:eagle"
+   account: candle_aesp_CNDA
+   queue: debug
+   walltime: 00:30:00
```
Comment (braceal) [inline examples/example.yaml]: Can you please make another yaml file, e.g. `examples/sunspot_example.yaml`. So we can keep the previous example as is? If we can change the name of the previous example to `examples/polaris_example.yaml` that would be great!

File: mdensemble/utils.py (hunk 1 above)
Comment (architvasan) [reply mdensemble/utils.py]: Sure, I'll remove this print

File: examples/example.yaml (hunk 3 above)
Comment (architvasan) [reply examples/example.yaml]: Oh sure! I'll do that.

File: mdensemble/parsl.py (hunk 2 above)
Comment (architvasan) [reply mdensemble/parsl.py]: So I wanted to keep these flags in the yaml file so they can be more modular. Similar to how it is for the polaris settings. I can remove these here

File: examples/example_sunspot.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  temperature_kelvin: 310
+  heat_bath_friction_coef: 1
+  explicit_barostat: MonteCarloAnisotropicBarostat
+
+# How many parallel tasks to run, this should generally be set to the number of GPUs available.
+num_parallel_tasks: 12
+
+# The compute settings
+
+# Uncomment this to run on Sunspot (please adjust settings appropriately
```
Comment (braceal) [inline examples/example_sunspot.yaml:25]: We can remove this comment in place of `Compute settings for sunspot`

File: examples/example_sunspot.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+num_parallel_tasks: 12
+
+# The compute settings
+
+# Uncomment this to run on Sunspot (please adjust settings appropriately 
+# and comment out the workstation compute settings)
+compute_settings:
+   name: sunspot # Run on the Polaris supercomputer @ ALCF
+   num_nodes: 1 # 4 GPUs per node, if you change this, make sure to change num_parallel_tasks
+   worker_init: "module load frameworks/2024.1; conda activate my_env"
```
Comment (braceal) [inline examples/example_sunspot.yaml:30]: Here it's surprising to me that we don't need to load more modules, export envs, etc. Nice that it is simple!

File: examples/example_sunspot.yaml (hunk 4 above)
Comment (architvasan) [reply examples/example_sunspot.yaml:25]: Sounds good

**Pull request merged**